x/merkledb/db.go

```diff
@@ -244,8 +244,13 @@ type merkleDB struct {
 }
 
 // New returns a new merkle database.
-func New(ctx context.Context, db database.Database, config Config) (MerkleDB, error) {
-	metrics, err := newMetrics("merkledb", config.Reg)
+func New(
+	ctx context.Context,
+	db database.Database,
+	config Config,
+	namespace string,
+) (MerkleDB, error) {
+	metrics, err := newMetrics(namespace, config.Reg)
```

Comment:
Should all of the metrics be prefixed with `merkledb` so that passing in `""` as the namespace is the same behavior as before?

Reply:
We could maintain the previous behavior but I don't see a reason to because nothing is using `merkledb` at the moment. I also feel like the `merkledb` prefix/being forced into the metric name is gross, it feels like it bleeds the implementation into the metric.